.circleci/config.yml

```diff
@@ -1,49 +1,68 @@
-version: 2
+version: 2.1
 
-test-template: &test-template
-    steps:
-        - checkout
-        - run:
-                name: environment
-                command: |
-                    echo 'export PATH=.:$HOME/.local/bin:$PATH' >> $BASH_ENV
-        - run:
-                name: install
-                command: |
-                    pip install --progress-bar off --user -U -r requirements_dev.txt
-                    sudo apt update -q
-                    sudo apt upgrade -q
-                    sudo apt install openjdk-11-jdk-headless
-                    make install-kotlin-linters
-
-        - run:
-                name: lint
-                command: make lint
-        - run:
-                name: install
-                command: python setup.py install --user
-        - run:
-                name: test
-                command: make test
-        - run:
-                name: make-docs
-                command: |
-                    make docs
-                    touch docs/_build/html/.nojekyll
-        - persist-to-workspace:
-                root: docs/_build
-                paths: html
+commands:
+    test-python-version:
+        parameters:
+            requirements-file:
+                type: string
+                default: "requirements_dev.txt"
+        steps:
+            - checkout
+            - run:
+                    name: environment
+                    command: |
+                        echo 'export PATH=.:$HOME/.local/bin:$PATH' >> $BASH_ENV
+            - run:
+                    name: install
+                    command: |
+                        pip install --progress-bar off --user -U -r <<parameters.requirements-file>>
+                        sudo apt update -q
+                        sudo apt upgrade -q
+                        sudo apt install openjdk-11-jdk-headless
+                        make install-kotlin-linters
+            - run:
+                    name: lint
+                    command: make lint
+            - run:
+                    name: install
+                    command: python setup.py install --user
+            - run:
+                    name: test
+                    command: make test
 
 jobs:
+    build-35:
+        docker:
+            - image: circleci/python:3.5.9
```

Comment:
The version used on Mozilla build machines is [Python 3.5.3](https://bugzilla.mozilla.org/show_bug.cgi?id=1598350#c5). Does it make sense to use that specific image?

Reply:
Python is generally pretty good about not breaking things with micro releases, but, sure, it probably makes sense to use the exact version for good measure.

Reply:
It turns out it was kind of nightmare to test against 3.5.3.  The 3.5.3 CircleCI Docker image has an entirely different (older) version of Debian, so the version of Java available is too old to run ktlint and detekt, and the version of `pip` is too old to install dependencies from a file.  I'm quite confident in Python that there weren't any breaking changes between 3.5.3 and 3.5.9, and it's easier to keep the rest of the non-Python part of our CI the same across all Python versions.

So I think we keep this as-is on 3.5.9 and re-address this if we run into issues trying to use it in m-c (which we should know immediately before merging into the m-c tree so is pretty low risk).

Reply:
Sounds good to me!